compiler/rustc_trait_selection/src/traits/query/type_op/mod.rs

```diff
@@ -190,10 +190,9 @@ where
                 }
             }
             if !progress {
-                return Err(infcx.dcx().span_delayed_bug(
-                    span,
-                    format!("ambiguity processing {obligations:?} from {self:?}"),
-                ));
+                infcx
+                    .dcx()
+                    .span_bug(span, format!("ambiguity processing {obligations:?} from {self:?}"));
```

Comment:
i think it's currently unreachable, but that may change going forward. Still Making it a bug seems alright to me